### PR TITLE
Elm 0.18 upgrade and some minor cleanups

### PR DESCRIPTION
**Pull request by clojj**
Hi, I'd like to help with the 0.18 upgrade
I tested all examples and ran all tests successfully

Regards
Joerg

### CODE REVIEW

**Comment by clojj** [comment]
Didn't see that other PR ..anyway... it's a choice now :)

**Comment by turboMaCk** [comment]
Hello @clojj 

Thank you for PR. Unfortunately there is already on upgrade PR opened: https://github.com/turboMaCk/chae-tree/pull/8

I'll look at yours one anyway.

File: examples/DummyExample.elm [hunk 1]
```diff
@@ -4,22 +4,24 @@ import List exposing (..)
...
 
+list : List { id : number, pid : List number1 }
 list =
     [ { id = 1, pid = [] }, { id = 2, pid = [] }, { id = 3, pid = [ 1 ] } ]
 
 
+tree2 : Tree.Tree { id : number, pid : List number1 }
 tree2 =
-    Tree.fromList (\a -> Id.toId (.id a)) (\a -> .pid a |> map Id.toId) list
+    Tree.fromList (\a -> Id.toId (.id a)) (\a -> .pid a |> List.map Id.toId) list
```
Comment (turboMaCk) [inline examples/DummyExample.elm:26]: 👍 These are good. Maybe we can put both upgrade PRs together since there are some interesting improvements in both.

**Comment by clojj** [comment]
Ok... btw,  I forgot the Travis upgrade

**Comment by turboMaCk** [comment]
@clojj no problem. I'll try to put both PRs together so you'll be both mentioned as contributors;) I let you know.

**Comment by clojj** [comment]
don't know Travis well enough, but running elm-test is fine on my machine
(upgraded the tests also to 0.18)

**Comment by clojj** [comment]
Oh I get it.. have to update package.json too

**Comment by turboMaCk** [comment]
@clojj Travis is using `elm-test` version installed via npm. Please update https://github.com/turboMaCk/chae-tree/blob/master/package.json#L12

you can also test it on your local machine by running
```
npm install && npm test
```

**Comment by turboMaCk** [comment]
@clojj 🙌 Nice. I want to merge other PR first or use it as base for merging your code. I'll let you know. Thanks for great work!

**Comment by clojj** [comment]
thx for this great package and the examples

**Comment by turboMaCk** [comment]
@clojj Feel free to rebase against master. I see there is one conflict in `tests/elm-package.json` hopefully it will be easy to resolve. I think we can get at least some goodness from your PR like additional type declaration in examples. 🚀

**Comment by turboMaCk** [comment]
```
git remote add upstream git@github.com:turboMaCk/chae-tree.git
git pull upstream master
```

should do the trik

**Comment by turboMaCk** [comment]
@clojj There you go: https://github.com/turboMaCk/chae-tree/commit/62d764e243f776ab929932cec9cfd46972fce415

Thanks again! Have a nice weekend!
🍺 time

**Comment by clojj** [comment]
sure !
now on to http://package.elm-lang.org
:)

**Comment by clojj** [comment]
(which makes me wonder if there is anything like a 'local' package-catalog ...?)
screen shot 2016-11-25 at 21 08 01

**Comment by turboMaCk** [comment]
@clojj 1.0.3 published https://github.com/turboMaCk/chae-tree/releases/tag/1.0.3

Usually if I need something which is not available in package.elm-lang.org I use git submodule.

**Comment by turboMaCk** [comment]
You're welcome! 
screen shot 2016-11-25 at 21 26 46

**Pull request merged**